mythril/laser/smt/bitvec.py

```diff
@@ -160,14 +181,340 @@ def __ne__(self, other: Union[int, "BitVec"]) -> Bool:  # type: ignore
         :param other:
         :return:
         """
+        if isinstance(other, BitVecFunc):
+            return other != self
         if not isinstance(other, BitVec):
-            return Bool(cast(z3.BoolRef, self.raw != other), annotations=self.annotations)
+            return Bool(
+                cast(z3.BoolRef, self.raw != other), annotations=self.annotations
+            )
 
         union = self.annotations + other.annotations
         # MYPY: fix complaints due to z3 overriding __eq__
         return Bool(cast(z3.BoolRef, self.raw != other.raw), annotations=union)
 
 
+class BitVecFunc(BitVec):
+    """A bit vector function symbol. Used in place of functions like sha3."""
+
+    def __init__(
+        self,
+        raw: z3.BitVecRef,
+        func_name: Optional[str],
+        input: Union[int, "BitVec"] = None,
+        annotations: Optional[Annotations] = None,
+    ):
+        """
+
+        :param raw: The raw bit vector symbol
+        :param func_name: The function name. e.g. sha3
+        :param input: The input to the functions
+        :param annotations: The annotations the BitVecFunc should start with
+        """
+
+        self.func_name = func_name
+        self.input = input
+        super().__init__(raw, annotations)
+
+    def __add__(self, other: Union[int, "BitVec"]) -> "BitVecFunc":
+        """Create an addition expression.
+
+        :param other:
+        :return:
+        """
+        if not isinstance(other, BitVec):
+            other = BitVec(z3.BitVecVal(other, self.size()))
+
+        raw = self.raw + other.raw
+        union = self.annotations + other.annotations
+
+        if isinstance(other, BitVecFunc):
+            # TODO: Find better value to set input and name to in this case
+            return BitVecFunc(raw=raw, func_name=None, input=None, annotations=union)
+
+        return BitVecFunc(raw=raw, func_name=self.func_name, input=self.input, annotations=union)
+
+    def __sub__(self, other: Union[int, "BitVec"]) -> "BitVecFunc":
```

Comment:
It seems to me that a lot of the logic for each of these implementations is equivalent, putting up a helper function that deals with everything and only passing it the operator could make this pr a bit shorter :+1: